scripts/idl/test_xml_parser.py

```diff
@@ -0,0 +1,197 @@
+                    <description>Reply from server</description>
+                    <arg name="dataPoint1" type="INT8U" />
+                    <arg name="dataPoint2" type="INT8U" optional="true" />
+                </command>
+              </cluster>
+            </configurator>
+        ''')
+        self.assertEqual(idl,
+                         Idl(clusters=[
+                             Cluster(
+                                 side=ClusterSide.CLIENT,
+                                 name='TestCluster',
+                                 code=0x1234,
+                                 attributes=[
+                                     Attribute(definition=Field(data_type=DataType(name='INT32U'), code=11, name='SomeIntAttribute',
+                                                                attributes={FieldAttribute.NULLABLE}), tags={AttributeTag.READABLE},
+                                               readacl=AccessPrivilege.VIEW, writeacl=AccessPrivilege.OPERATE),
+                                     Attribute(definition=Field(data_type=DataType(name='INT8U'), code=22, name='AttributeWithAccess',
+                                                                attributes={FieldAttribute.OPTIONAL}),
+                                               tags={AttributeTag.READABLE, AttributeTag.WRITABLE}, readacl=AccessPrivilege.OPERATE,
+                                               writeacl=AccessPrivilege.MANAGE)
+                                 ],
+                                 structs=[
+                                     Struct(name='GetSomeDataRequest',
+                                            fields=[
+                                                Field(data_type=DataType(name='INT8U'), code=1, name='firstInput'),
+                                                Field(data_type=DataType(name='INT16U'), code=2, name='secondInput')
+                                            ],
+                                            tag=StructTag.REQUEST),
+                                     Struct(name='GetSomeDataResponse',
+                                            fields=[
+                                                Field(data_type=DataType(name='INT8U'), code=1,
+                                                      name='dataPoint1'),
+                                                Field(data_type=DataType(name='INT8U'), code=2, name='dataPoint2',
+                                                      attributes={FieldAttribute.OPTIONAL})
+                                            ],
+                                            tag=StructTag.RESPONSE, code=0x44)
+                                 ],
+                                 commands=[
+                                     Command(name='GetSomeData', code=33, input_param='GetSomeDataRequest', output_param='GetSomeDataResponse',
```

Comment:
I think it's not quite aligned with the Matter data model to state this combination of request and response pair as a 'command'.

Instead, it's better to annotate the CommandRequest with a field containing a pointer to its logical response.

Reply:
Matching current IDL. Re-formatting of the structure should be a followup.

Also as a separate: all RPC systems I know of are  'a RPC has a request and a response'. I would like to have IDL feel familiar as long as this can meet spec definitions. Existing logic of 'client side', 'server side', client commands, server commands is alien and we should try to make matter accessible.